src/qgraphicsmozview_p.cpp

```diff
@@ -295,10 +303,7 @@ void QGraphicsMozViewPrivate::OnLoadStarted(const char* aLocation)
 {
     Q_UNUSED(aLocation);
 
-    if (mIsPainted) {
-        mIsPainted = false;
-        mViewIface->firstPaint(-1, -1);
-    }
+    ResetPainted();
```

Comment:
Hmm, I'm not sure we should really do both, I think we should call mViewIface->firstPaint only from OnFirstPaint method, which get's called right after gecko done with first page paint, if that does not work very well then we should fix it somewhere in https://github.com/tmeshkova/gecko-dev/blob/embedlite/embedding/embedlite/utils/WebBrowserChrome.cpp#L423
you need this checking state when you can or cannot grab screen shots, and I see that all these changes are doable outside of qtmozembed using existing API.

I would suggest to remove  298-301 from here, and move resetPainted logic to QML, using loadingChanged , firstPaint signals and navigation method calls.